app/views/Partners/PartnersCustomUrlScreen.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState, useCallback, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { StyleSheet, View, TextInput, Alert } from 'react-native';
+import {
+  NavigationParams,
+  NavigationScreenProp,
+  NavigationState,
+} from 'react-navigation';
+
+import { NavigationBarWrapper } from '../../components/NavigationBarWrapper';
+
+import { useDispatch, useSelector } from 'react-redux';
+import getHealthcareAuthoritiesAction from '../../store/actions/healthcareAuthorities/getHealthcareAuthoritiesAction';
+import { Typography } from '../../components/Typography';
+
+import { Button } from '../../components/Button';
+import { Colors } from '../../styles';
+
+type PartnersEditScreenProps = {
+  navigation: NavigationScreenProp<NavigationState, NavigationParams>;
+};
+
+const PartnersScreen = ({
+  navigation,
+}: PartnersEditScreenProps): JSX.Element => {
+  const { t } = useTranslation();
+  const [url, setUrl] = useState('');
+  const didMountRef = useRef(false);
+
+  const dispatch = useDispatch();
+
+  // This is testing code only,we omit selector:
+  const fetchCustomStatus = useSelector(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (state: any) => state.healthcareAuthorities.request.status,
+  );
+
+  const onRequestComplete = useCallback(() => {
+    // We only care if this happened from a trigger on this screen,
+    // not if we enter this screen on this state. Hack to achieve this, not
+    // super robust against parallel events, but good for this screen's testing purposes.
+    if (didMountRef.current) {
```

Comment:
ah, i see the reason for the ref here.  does this feature want to live in a parent component then?  Whats the need to be using memoization at all? could we get away with the same functionality without introducing a ref/callback hook if we dont have these side effects coupled to the screen components render cycle?

Reply:
The goal here is achieve a `useEffect_exceptOnFirstMount` behavior. It needs to be at the same level as the redux selector logic. 
An alternative approach would be a `usePrev` state hook for mocking a more "componentDidUpdate" lifecycle, or building a explicit `useEffect_exceptOnFirstMount` hook. 

I went with this for the most lightweight solution here, since this screen is just for testing

Reply:
got it yeah, i wouldnt spend too much time on it, but it is quite a complicated thing which jumps out to me as a place for confusion/bugs.